cmake/kokkos_arch.cmake

```diff
@@ -571,6 +571,9 @@ IF (KOKKOS_ENABLE_HIP)
     COMPILER_SPECIFIC_FLAGS(
       DEFAULT -fgpu-rdc
     )
+    COMPILER_SPECIFIC_LINK_OPTIONS(
```

Comment:
@Rombur  What's the best way to do this?
* Comment + add unconditionally
* Put the call within a `IF(NOT KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC)`
* Specify a non-default clause for hipcc with an empty flag?

Reply:
First, I want to try on Frontier. We have seen in the past the cray compiler behaving differently on different machines. If it works correctly,  we should guard this flag with `IF(NOT KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC)`